src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt

```diff
@@ -16,7 +17,8 @@ import java.util.function.Consumer
 class CodyAgentService(project: Project) : Disposable {
 
   private val logger = Logger.getInstance(CodyAgent::class.java)
-  private var codyAgent: CompletableFuture<CodyAgent> = CompletableFuture()
+  private var codyAgent: AtomicReference<CompletableFuture<CodyAgent>> =
```

Comment:
It doesn't feel like `AtomicReference` is doing much work here when we're not using operators like compareAndSet. I suppose it's working similar to a `volatile` variable.